### DISCUSSION TITLE
OIDC group setting

### DISCUSSION THREAD

**Discussion by drustan** [peering-manager/peering-manager#868] [General]
Hi everyone,

I've successfully set up SSO with Keycloak using the documentation provided here: https://peering-manager.readthedocs.io/en/stable/setup/5b-oidc/. Everything is working perfectly.

However, I noticed that there's no mention on how to map the groups from the access token to automatically assign new users to the correct Peering Manager group. Currently, after a user is automatically created via OpenID connection, I still need to manually place them in the appropriate group.

Does anyone know how to configure this to automate group assignments?

Thanks in advance!

**Comment by gmazoyer** [comment]
There are no ways to do this in stable. However there is a major refactoring of user authentication coming in 1.9. This will probably also needs polishing for automatic group assignments though. Just in case, you should be able to review the doc already by looking at https://peering-manager.readthedocs.io/en/latest/.

**Reply by drustan** [reply]
Thanks for your quick answer and all the hard work you're putting into this. I guess I'm looking forward to the upcoming version now :-)

**Comment by gmazoyer** [comment]
To keep you posted, [I just added an example in the documentation of an OIDC configuration](https://github.com/peering-manager/peering-manager/commit/085f19e12f4b755c1a2205da2818ac5d6d51c3af) that should work with Peering Manager when 1.9.0 will be released. I used Authentik as an authentication backend, but I guess you'll be able to figure out how to adapt the config to your own needs.

**Reply by drustan** [reply]
Thanks for letting me know. I'll look at the new version of the doc :)
